src/main/java/pl/nogacz/checkers/board/Board.java

```diff
@@ -404,4 +476,129 @@ public static boolean isThisSameColor(Coordinates coordinates, PawnColor color)
+            setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+            this.addWindowListener(new WindowAdapter() {
+                public void windowClosing(WindowEvent e) {
+                     isEditMenuActive = false;
+                }
+            });
+            setLayout(null);
+            setVisible(true);
+            setLocation(1330,200);
+            toFront();
+        }
+
+        private void setButtons() {
+            String removeButtonPNG = this.getClass().getClassLoader().getResource("button_remove-selected-pawn.png").getFile();
+            removeButton.setIcon(new ImageIcon(removeButtonPNG));
+            removeButton.setBorder(BorderFactory.createEmptyBorder());
+            removeButton.setContentAreaFilled(false);
+            removeButton.setFocusable(false);
+            String replaceButtonPNG = this.getClass().getClassLoader().getResource("button_replace-selected-pawn.png").getFile();
+            replaceButton.setIcon(new ImageIcon(replaceButtonPNG));
+            replaceButton.setBorder(BorderFactory.createEmptyBorder());
+            replaceButton.setContentAreaFilled(false);
+            replaceButton.setFocusable(false);
+            String exitButtonPNG = this.getClass().getClassLoader().getResource("button_ready.png").getFile();
+            exitButton.setIcon(new ImageIcon(exitButtonPNG));
+            exitButton.setBorder(BorderFactory.createEmptyBorder());
+            exitButton.setContentAreaFilled(false);
+            exitButton.setFocusable(false);
+            String addWhiteButtonPNG = this.getClass().getClassLoader().getResource("button_add-white-pawn.png").getFile();
+            addWhitePawnButton.setIcon(new ImageIcon(addWhiteButtonPNG));
+            addWhitePawnButton.setBorder(BorderFactory.createEmptyBorder());
+            addWhitePawnButton.setContentAreaFilled(false);
+            addWhitePawnButton.setFocusable(false);
+            String addBlackButtonPNG = this.getClass().getClassLoader().getResource("button_add-black-pawn.png").getFile();
+            addBlackPawnButton.setIcon(new ImageIcon(addBlackButtonPNG));
+            addBlackPawnButton.setBorder(BorderFactory.createEmptyBorder());
+            addBlackPawnButton.setContentAreaFilled(false);
+            addBlackPawnButton.setFocusable(false);
+            removeButton.setBounds(20, 20, 300, 60);
+            replaceButton.setBounds(20, 100, 300, 60);
+            addWhitePawnButton.setBounds(20,180,300,60);
+            addBlackPawnButton.setBounds(20,260,300,60);
+            exitButton.setBounds(20, 360, 310, 60);
+            addWhitePawnButton.addActionListener(this);
+            addBlackPawnButton.addActionListener(this);
+            removeButton.addActionListener(this);
+            replaceButton.addActionListener(this);
+            exitButton.addActionListener(this);
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            String command = e.getActionCommand();

```

Comment:
Bu kısmı başka bir class içerisinde implement et ve ActionEvent i o classın objesine delegate etç

Böylece pencerenin görsel implementasyonu ile kullanıcı inputlarına nasıl reaksiyon verdiğini ayırmış olursun: (bakınız separation of concerns)